shared/js/background/tracker-utils.js

```diff
@@ -30,6 +31,114 @@ function isTracker (url) {
     return !!tracker
 }
 
+/**
+ * Determine if the social entity should be blockedon this URL. returns True if so.
+ */
+function shouldBlockSocialNetwork (entity, url) {
+    const domain = tldts.parse(url).domain
+    const excludeData = getDomainsToExludeByNetwork()
+    return excludeData.filter(e => e.domain === domain && e.entity === entity).length === 0
+}
+
+/*
+ * Parse the social config to find excluded domains by social tracker. This then returns a list of objects
+ * that include the exlcuded domain and network, for use in other exception list handling.
+ */
+function getDomainsToExludeByNetwork () {
+    const excludes = []
+    for (const [entity, data] of Object.entries(tdsStorage.ClickToLoadConfig)) {
+        if (data.excludedDomains) {
+            const excludedDomains = data.excludedDomains.map(e => e.domain)
+            for (const domain of excludedDomains) {
+                excludes.push({
+                    entity: entity,
+                    domain: domain
+                })
+            }
+        }
+    }
+    return excludes
+}
+
+// Return true if URL is in our click to load tracker list
+function getSocialTracker (url) {
+    if (!facebookExperimentIsActive()) {
+        return
+    }
+    const parsedDomain = tldts.parse(url)
+    for (const [entity, data] of Object.entries(tdsStorage.ClickToLoadConfig)) {
+        if (data.domains.includes(parsedDomain.domain) && !data.excludedSubdomains.includes(parsedDomain.hostname)) {
+            let redirect
+            if (data.surrogates) {
+                for (const surrogate of data.surrogates) {
+                    if (url.match(surrogate.rule)) {
+                        redirect = surrogate.surrogate
```

Comment:
In the case of multiple matching rules, the _last_ one will be used. Is that intended or should there be a break here?

Reply:
Good thinking!